### PR TITLE
Validate configurations and add configuration tests

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: app/src/main/java/com/github/bsideup/liiklus/Application.java [hunk 1]
```diff
@@ -58,8 +58,8 @@ public static SpringApplication createSpringApplication(String[] args) {
         environment.setDefaultProfiles("exporter", "gateway");
         environment.getPropertySources().addFirst(new SimpleCommandLinePropertySource(args));
 
-        var pluginsDir = environment.getProperty("plugins.dir", String.class, "./plugins");
-        var pathMatcher = environment.getProperty("plugins.pathMatcher", String.class, "*.jar");
+        var pluginsDir = environment.getRequiredProperty("plugins.dir", String.class);
```
Comment (lanwen) [inline app/src/main/java/com/github/bsideup/liiklus/Application.java]: why without default now?

File: app/src/main/java/com/github/bsideup/liiklus/config/GatewayConfiguration.java [hunk 2]
```diff
@@ -46,6 +46,7 @@ public void initialize(GenericApplicationContext applicationContext) {
         applicationContext.registerBean(LiiklusService.class);
     }
 
+    @ConfigurationProperties("layers")
```
Comment (lanwen) [inline app/src/main/java/com/github/bsideup/liiklus/config/GatewayConfiguration.java]: get back to annotations?

File: app/src/main/java/com/github/bsideup/liiklus/Application.java (hunk 1 above)
Comment (lanwen) [reply app/src/main/java/com/github/bsideup/liiklus/Application.java]: ah, moved to config

File: plugins/grpc-transport/build.gradle [hunk 3]
```diff
@@ -74,4 +74,14 @@ dependencies {
     compile 'io.grpc:grpc-protobuf'
     compile 'io.grpc:grpc-netty'
     compile 'io.grpc:grpc-services'
+
+    testCompileOnly 'org.projectlombok:lombok'
+    testAnnotationProcessor 'org.projectlombok:lombok'
+    testCompile 'org.junit.jupiter:junit-jupiter-api'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
```
Comment (lanwen) [inline plugins/grpc-transport/build.gradle]: isn't that better to move on top-level and define for all subprojects?

File: app/src/main/java/com/github/bsideup/liiklus/config/GatewayConfiguration.java (hunk 2 above)
Comment (bsideup) [reply app/src/main/java/com/github/bsideup/liiklus/config/GatewayConfiguration.java]: yeah, everything in one place, and to have less differences with the classical boot app

File: plugins/grpc-transport/build.gradle (hunk 3 above)
Comment (bsideup) [reply plugins/grpc-transport/build.gradle]: not until all subprojects use jupiterr

File: app/src/main/java/com/github/bsideup/liiklus/Application.java (hunk 1 above)
Comment (bsideup) [reply app/src/main/java/com/github/bsideup/liiklus/Application.java]: yeah, wanted to not have the defaults in code

**Pull request merged**